### PR TITLE
server: Do not resize volume of running vm on KVM host if host is not Up or not Enabled

### PR DESCRIPTION
**Pull request by ustcweizhou**
##  Description

If we resize a volume of a vm running on a host which is not Up or not Enable, the job will be scheduled to another normal host. Then the volume will be resized by "qemu-img resize" instead of "virsh blockresize", the image might be corrupted after resize.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

File: server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java [hunk 1]
```diff
@@ -1187,6 +1189,16 @@ private VolumeVO orchestrateResizeVolume(long volumeId, long currentSize, long n
             if (currentSize != newSize && _volsDao.getHypervisorType(volume.getId()) == HypervisorType.XenServer && !userVm.getState().equals(State.Stopped)) {
                 throw new InvalidParameterValueException(errorMsg);
             }
+
+            /* Do not resize volume of running vm on KVM host if host is not Up or not Enabled */
+            if (currentSize != newSize && userVm.getState() == State.Running && userVm.getHypervisorType() == HypervisorType.KVM) {
+                HostVO host = _hostDao.findById(userVm.getHostId());
+                if (host.getStatus() != Status.Up) {
```
Comment (rhtyd) [inline server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java:1196]: @ustcweizhou should you do a null check, what if the host was removed?

Comment (weizhouapache) [reply server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java]: @rhtyd added null check

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
LGTM

**Pull request merged**